### PR TITLE
Finalizing Reject and Accept View and Delegates

### PR DESCRIPTION
**Pull request by Perklone**
aku gk tau ada mem issue gk dari observer yang dibuat sama target2nya pas keluar dari view controller tapi double check aja deh, kurang tau deinitnya jalan atau nggak itu.

### CODE REVIEW

File: Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift [hunk 1]
```diff
@@ -253,16 +261,20 @@ class EventDetailContainerViewController: UIViewController, UIScrollViewDelegate
                 DispatchQueue.main.async {
                     self.joinButtonView = RejectAcceptView()
```
Comment (gremlinflat) [inline Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift:262]: change name to generic name e.g BOTTOMPOPUPView

File: Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift [hunk 2]
```diff
@@ -253,16 +261,20 @@ class EventDetailContainerViewController: UIViewController, UIScrollViewDelegate
...
                     self.configureJoinView(for: 0)
-                    (self.joinButtonView as! RejectAcceptView).rejectButton.addTarget(self, action: #selector(self.test(_:)), for: .touchUpInside)
-                    (self.joinButtonView as! RejectAcceptView).acceptButton.addTarget(self, action: #selector(self.test(_:)), for: .touchUpInside)
+                    (self.joinButtonView as! RejectAcceptView).rejectButton.addTarget(self, action: #selector(self.rejectButtonTapped(_:)), for: .touchUpInside)
```
Comment (gremlinflat) [inline Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift:264]: move these closures to a specific function e.g. configure popupAsAcceptReject

File: Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift [hunk 3]
```diff
@@ -253,16 +261,20 @@ class EventDetailContainerViewController: UIViewController, UIScrollViewDelegate
...
+                    (self.joinButtonView as! RejectAcceptView).saveButton.addTarget(self, action: #selector(self.saveButtonPressed(_:)), for: .touchUpInside)
+                    self.enableJoinButton
+                        .receive(on: RunLoop.main)
+                        .assign(to: \.isEnabled, on: (self.joinButtonView as! RejectAcceptView).saveButton)
+                        .store(in: &self.cancellable)
                 }
             } else {
                 DispatchQueue.main.async {
                     self.joinButtonView = CurrentParticipantAndJoinView()
                     self.configureJoinView(for: 1)
```
Comment (gremlinflat) [inline Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift:275]: same as above

File: Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift [hunk 4]
```diff
@@ -278,7 +290,50 @@ class EventDetailContainerViewController: UIViewController, UIScrollViewDelegate
         ])
     }
     
-    @objc func test(_ sender: UIButton) {
+    //MARK: - Join Button Actions
+    
+    @objc func rejectButtonTapped(_ sender: UIButton) {
```
Comment (gremlinflat) [inline Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift:295]: did____

**Review by gremlinflat — CHANGES_REQUESTED** [review]
request changes

**Comment by Perklone** [comment]
fixed #84

**Review by gremlinflat — CHANGES_REQUESTED** [review]
ada probability pindah screen jdi add target berkali kali (ARC telat release memory).
preventive solution : add remove target at didDissapear lifecycle & move fetch invite to didappear.

File: Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift [hunk 5]
```diff
@@ -298,6 +299,14 @@ class EventDetailContainerViewController: UIViewController, UIScrollViewDelegate
         }
     }
     
+    func detachTargets() {
```
Comment (gremlinflat) [inline Sounds Like A Plan/Features/Event Details V2/EventDetailContainerViewController.swift:302]: jgn di forcecast klo disini. pake optional would be safer
  (bottomPopUpView as? RejectAcceptView)?.rejectButton
  (bottomPopUpView as? CurrentParticipantAndJoinView)?.joinButton

**Comment by Perklone** [comment]
fixed #84

**Review by gremlinflat — APPROVED** [review]
LGTM

**Pull request merged**